pkg/liqoctl/install/utils/map.go

```diff
@@ -0,0 +1,53 @@
+package installutils
+
+import (
+	"fmt"
+	"reflect"
+)
+
+// FusionMap fusions two maps recursively writing the result in expectedResultMap result passed as argument. In case of duplicated keys,
+// the values extracted from patchMap are considered valid.
+func FusionMap(baseMap, patchMap, resultMap map[string]interface{}) error {
+	for _, key := range extractKeys(baseMap, patchMap) {
+		v, ok := baseMap[key]
+		v2, ok2 := patchMap[key]
+
+		if ok && !ok2 {
+			resultMap[key] = v
+			continue
+		} else if !ok && ok2 {
+			resultMap[key] = v2
+			continue
+		}
+
+		switch {
+		case reflect.TypeOf(v) != reflect.TypeOf(v2):
+			return fmt.Errorf("the two maps have different types for the same key")
+		case reflect.TypeOf(v).String() == "string", reflect.TypeOf(v).String() == "bool", reflect.TypeOf(v).String() == "int":
+			resultMap[key] = v2
+		default:
+			resultMap[key] = make(map[string]interface{})
+			err := FusionMap(baseMap[key].(map[string]interface{}), patchMap[key].(map[string]interface{}), resultMap[key].(map[string]interface{}))
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func extractKeys(baseMap, patchMap map[string]interface{}) []string {
+	keys := make([]string, len(baseMap)+len(patchMap))
+	i := 0
+	for k := range baseMap {
+		keys[i] = k
+		i++
+	}
+
+	for k := range patchMap {
+		keys[i] = k
+		i++
+	}
+	return keys
```

Comment:
here there can be a lot of repeated values, I can suggest something like that:

```suggestion
	keys := make(map[string]struct{})
	for k := range baseMap {
		keys[k] = struct{}
	}

	for k := range patchMap {
		keys[k] = struct{}
	}
	
	keysArr := make([]string, len(keys))
	i := 0
	for k := range keys {
		keys[i] = k
		i++
	}
	return keysArr
```